cs/src/Contracts/TunnelEndpoint.cs

```diff
@@ -94,6 +102,31 @@ public abstract class TunnelEndpoint
             PortUriToken, portNumber.Value.ToString(CultureInfo.InvariantCulture)));
     }
 
+    /// <summary>
+    /// Gets a ssh command which can be used to connect to a tunnel ssh port. 
+    /// </summary>
+    /// <param name="endpoint">A tunnel endpoint containing a port ssh URI format.</param>
+    /// <param name="portNumber">The port number to connect to; the port is assumed to be
+    /// separately shared by a tunnel host.</param>
+    /// <returns>ssh command for the requested ssh port, or null if the endpoint does not support
+    /// ssh client connections.</returns>
+    /// <remarks>
+    /// SSH client on Windows/Linux/MacOS  are supported.
+    /// <para />
+    /// If the port is not currently shared via the tunnel, or if a host is not currently
+    /// connected to the tunnel, then ssh connection might fail.
+    /// </remarks>
+    public static string GetPortSshCommand(TunnelEndpoint endpoint, int? portNumber)
+    {
+        if (portNumber == null || string.IsNullOrEmpty(endpoint.PortSshCommandFormat))
+        {
+            return string.Empty;
```

Comment:
For consistency with `GetPortUri()` maybe this should return `null` in this case. (And change return type to `string?`).